src/Analyzer.Reports/SarifReportWriter.cs

```diff
@@ -0,0 +1,195 @@
+                        },
+                        DefaultConfiguration = new ReportingConfiguration { Level = GetLevelFromEvaluation(evaluation) }
+                    });
+            }
+            return rulesDictionary[evaluation.RuleId];
+        }
+
+        private void ExtractResult(IEvaluation rootEvaluation, IEvaluation evaluation, string filePath)
+        {
+            foreach (var result in evaluation.Results.Where(r => !r.Passed))
+            {
+                this.sarifResults.Add(new Result
+                {
+                    RuleId = rootEvaluation.RuleId,
+                    Level = GetLevelFromEvaluation(rootEvaluation),
+                    Message = new Message { Id = "default" }, // should be customized message for each result 
+                    Locations = new[]
+                    {
+                        new Location
+                        {
+                            PhysicalLocation = new PhysicalLocation
+                            {
+                                ArtifactLocation = new ArtifactLocation
+                                {
+                                    Uri = new Uri(
+                                    UriHelper.MakeValidUri(
+                                        filePath.Replace(this.rootPath, string.Empty, StringComparison.OrdinalIgnoreCase)),
+                                    UriKind.Relative),
+                                    UriBaseId = UriBaseIdString,
+                                },
+                                Region = new Region { StartLine = result.LineNumber },
+                            },
+                        },
+                    }
+                });
+            }
+
+            foreach (var eval in evaluation.Evaluations.Where(e => !e.Passed))
+            {
+                this.ExtractResult(rootEvaluation, eval, filePath);
+            }
+        }
+
+        private FailureLevel GetLevelFromEvaluation(IEvaluation evaluation)
+        {
+            // The rule severity definition work item: https://github.com/Azure/template-analyzer/issues/177
+            return evaluation.Passed ? FailureLevel.Note : FailureLevel.Error;
+        }
+
+        private void PersistReport()
+        {
+            using Stream outputTextStream = this.reportFile.Create();
+            using var outputTextWriter = new StreamWriter(outputTextStream);
+            using var sarifLogger = new SarifLogger(
+                textWriter: outputTextWriter,
+                logFilePersistenceOptions: LogFilePersistenceOptions.PrettyPrint | LogFilePersistenceOptions.OverwriteExistingOutputFile,
+                tool: this.sarifRun.Tool,
+                run: this.sarifRun,
+                levels: new List<FailureLevel> { FailureLevel.Warning, FailureLevel.Error, FailureLevel.Note },
+                kinds: new List<ResultKind> { ResultKind.Fail, ResultKind.Pass });
```

Comment:
>Pass [](http://example.com/codeflow?start=74&length=4)

are u emitting a pass? if not, you can exclude.

Reply:
not emitting pass results, removed Pass ResultKind.